### PR TITLE
[#174574145] MyPortal auth spec

### PR DESCRIPTION
**Pull request by BurnedMarshal**
`getUser` API for MyPortal authentication.

### CODE REVIEW

**Comment by cloudify** [comment]
it looks like `MyPortalUser` is equivalent to a `PagopaUser`, perhaps we can define a `FederatedUser` that can be shared by both?

**Comment by BurnedMarshal** [comment]
> it looks like `MyPortalUser` is equivalent to a `PagopaUser`, perhaps we can define a `FederatedUser` that can be shared by both?

@cloudify Yes, we can define a `FederatedUser` with `name, surname and fiscal_code` and share it with `MyPortalUser` (identical) and `ExtendedPagopaUser`. `PagoPAUser` haven't the `fiscal_code` field that is required in both `MyPortalUser` and `ExtendedPagopaUser`.

File: api_myportal.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+security:
+  - Token: []
+paths:
+  "/user":
+    x-swagger-router-controller: MyPortalController
+    get:
+      operationId: getUser
+      summary: Get user's data
+      description: Returns the user data needed by MyPortal backend.
+      parameters:
```
Comment (francescopersico) [inline api_myportal.yaml:19]: Invalid, please remove.

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: api_myportal.yaml (hunk 1 above)
Comment (BurnedMarshal) [reply api_myportal.yaml]: Fixed in 42d4426

**Comment by francescopersico** [comment]
@BurnedMarshal @gunzip do we need to maintain both PagoPAUser and ExtendedPagoPAUser? Can we delete one, maintain only the updated version and simplify the pagoPAController code?

**Comment by BurnedMarshal** [comment]
> @BurnedMarshal @gunzip do we need to maintain both PagoPAUser and ExtendedPagoPAUser? Can we delete one, maintain only the updated version and simplify the pagoPAController code?

@francescopersico if the Payment Manager never uses the `getUser` without `version=20200114` we can dismiss the old `PagoPAUser`. Both versions were required only for the transition phase. I would do it on another PR.

**Comment by francescopersico** [comment]
> > @BurnedMarshal @gunzip do we need to maintain both PagoPAUser and ExtendedPagoPAUser? Can we delete one, maintain only the updated version and simplify the pagoPAController code?
> 
> @francescopersico if the Payment Manager never uses the `getUser` without `version=20200114` we can dismiss the old `PagoPAUser`. Both versions were required only for the transition phase. I would do it on another PR.

As you said it was only used in the transition phase. I'll check but i am quite sure we can just use a single PagoPAUser type(with all the field of ExtendedPagoPAUser type) and remove the version check.
Let's deal with the problem in another PR, i am going to create the story.

File: api_myportal.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  - Token: []
+paths:
+  "/user":
+    x-swagger-router-controller: MyPortalController
+    get:
+      operationId: getUser
+      summary: Get user's data
+      description: Returns the user data needed by MyPortal backend.
+      responses:
+        "200":
```
Comment (francescopersico) [inline api_myportal.yaml:20]: @BurnedMarshal 
Where is the "not found" case?

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: api_myportal.yaml (hunk 2 above)
Comment (BurnedMarshal) [reply api_myportal.yaml:20]: This API will use only the data stored in Redis. If the token is missing we are into the `401` scenario. `404` response occurs when the user profile is missing but it doesn't happen into this API.

**Review by francescopersico — APPROVED** [review]

**Comment by francescopersico** [comment]
@gunzip Do you think we can merge this?

File: api_myportal.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+              fiscal_code: "AAABBB01C02D123Z"
+        "401":
+          description: Token null or expired.
+        "500":
+          description: There was an error in retrieving the user data.
+          schema:
+            $ref: "#/definitions/ProblemJson"
+definitions:
+  MyPortalUser:
+    $ref: "https://raw.githubusercontent.com/pagopa/io-backend/174574145-myportal-auth-spec/api_backend.yaml#/definitions/FederatedUser"
```
Comment (gunzip) [inline api_myportal.yaml:37]: remember to change this pointer one merged

File: api_pagopa.yaml [hunk 4]
```diff
@@ -80,29 +80,24 @@ definitions:
       - family_name
       - email
       - mobile_phone
+  FederatedUser:
+    $ref: "https://raw.githubusercontent.com/pagopa/io-backend/174574145-myportal-auth-spec/api_backend.yaml#/definitions/FederatedUser"
```
Comment (gunzip) [inline api_pagopa.yaml:84]: ditto

**Comment by gunzip** [comment]
lgtm but we must change the link to the branch once merged

**Pull request merged**